pkg/controllers/monitoring/monitoring-stack/components.go

```diff
@@ -22,7 +23,14 @@ const AdditionalScrapeConfigsSelfScrapeKey = "self-scrape-config"
 const PrometheusUserFSGroupID = int64(65534)
 const AlertmanagerUserFSGroupID = int64(65535)
 
-func stackComponentReconcilers(ms *stack.MonitoringStack, instanceSelectorKey string, instanceSelectorValue string) []reconciler.Reconciler {
+func stackComponentReconcilers(
+	ms *stack.MonitoringStack,
+	instanceSelectorKey string,
+	instanceSelectorValue string,
+	thanos tqctrl.ThanosConfiguration,
```

Comment:
NIT thanosCfg, prometheusCfg and alertmanagerCfg might be more descriptive

Reply:
:+1: will address that after we agree on the general structure of this change.